### PR TITLE
Size matters

### PR DESCRIPTION
**Pull request by Shyrx**

### CODE REVIEW

File: src/process/process.cc [hunk 1]
```diff
@@ -77,45 +77,66 @@ namespace blind
             return sqrt(pow(a.x - b.x, 2) + pow(a.y - b.y, 2));
         }
 
-        cv::Point getDirection(const cv::Point &base, const int nb_labels,
-                               cv::Mat &centroids, cv::Mat &yellow_cones)
+        cv::Point getMiddle(cv::Point &red, cv::Point &yellow, int max_cols,
+                            int max_rows)
         {
+            cv::Point mid =
+                cv::Point((red.x + yellow.x) / 2, (red.y + yellow.y) / 2);
```
Comment (TristanFloch) [inline src/process/process.cc:84]:
```suggestion
            cv::Point mid((red.x + yellow.x) / 2, (red.y + yellow.y) / 2);
```

File: src/process/process.cc [hunk 2]
```diff
@@ -77,45 +77,66 @@ namespace blind
             return sqrt(pow(a.x - b.x, 2) + pow(a.y - b.y, 2));
         }
 
-        cv::Point getDirection(const cv::Point &base, const int nb_labels,
-                               cv::Mat &centroids, cv::Mat &yellow_cones)
+        cv::Point getMiddle(cv::Point &red, cv::Point &yellow, int max_cols,
+                            int max_rows)
```
Comment (TristanFloch) [inline src/process/process.cc:81]:
```suggestion
        cv::Point getMiddle(const cv::Point &red, const cv::Point &yellow, int max_cols,
                                            int max_rows)
```

File: src/process/process.cc [hunk 3]
```diff
@@ -77,45 +77,66 @@ namespace blind
...
+            int max_yellow_size = 0;
             for (int i = 1; i < nb_labels; i++)
             {
-                cv::Point p(centroids.at<double>(i, 0),
-                            centroids.at<double>(i, 1));
-                auto val = yellow_cones.at<unsigned char>(p);
+                cv::Point curr_p(centroids.at<double>(i, 0),
+                                 centroids.at<double>(i, 1));
+                auto val = yellow_cones.at<unsigned char>(curr_p);
+                int size = stats.at<int>(i, cv::CC_STAT_AREA);
```
Comment (TristanFloch) [inline src/process/process.cc:107]:
```suggestion
                auto size = stats.at<int>(i, cv::CC_STAT_AREA);
```

File: src/process/process.cc [hunk 4]
```diff
@@ -77,45 +77,66 @@ namespace blind
...
+
+                auto func = [&base, &curr_p, &size](int &max_size,
```
Comment (TristanFloch) [inline src/process/process.cc:109]: Please rename this 🥸

File: src/process/process.cc [hunk 5]
```diff
@@ -135,7 +156,7 @@ namespace blind
             all_cones, labels, stats, centroids);
 
         cv::Point base(img.cols / 2, img.rows);
-        cv::Point mid = getDirection(base, nb_labels, centroids, yellow_cones);
+        cv::Point mid = getDirection(nb_labels, centroids, yellow_cones, stats);
```
Comment (TristanFloch) [inline src/process/process.cc:159]:
```suggestion
        const auto mid = getDirection(nb_labels, centroids, yellow_cones, stats);
```

File: src/process/process.cc [hunk 6]
```diff
@@ -135,7 +156,7 @@ namespace blind
             all_cones, labels, stats, centroids);
 
         cv::Point base(img.cols / 2, img.rows);
```
Comment (TristanFloch) [inline src/process/process.cc:158]:
```suggestion
        const cv::Point base(img.cols / 2, img.rows);
```

**Review by TristanFloch — COMMENTED** [review]
Please check if the suggestions compile **and run** before merging (OpenCV sometimes fails with constness).

File: src/process/process.cc (hunk 3 above)
Comment (Shyrx) [reply src/process/process.cc:107]: Why would auto be better than specifying the type for a simple one as 'int' ?

File: src/process/process.cc (hunk 4 above)
Comment (Shyrx) [reply src/process/process.cc:109]: I don't have any name idea ! But I agree that we can't let `func` as name ^^

File: src/process/process.cc (hunk 3 above)
Comment (TristanFloch) [reply src/process/process.cc:106]: Consistency

File: src/process/process.cc (hunk 4 above)
Comment (TristanFloch) [reply src/process/process.cc:108]: I though about it but I had trouble understanding what the function was doing so it complicates things

**Review by Cobaltarena — APPROVED** [review]

**Review by TristanFloch — APPROVED** [review]

**Pull request merged**